packages/esm-ward-app/src/ward-view/ward-view.resource.test.ts

```diff
@@ -0,0 +1,140 @@
+import { createAndGetWardPatientGrouping, getWardMetrics } from './ward-view.resource';
+import { type AdmissionLocationFetchResponse, type InpatientAdmission, type WardPatientGroupDetails } from '../types';
+import type { Location } from '@openmrs/esm-framework';
+
+describe('Ward View Resource', () => {
+  const patient1 = {
+    uuid: 'patient-uuid-1',
+    person: {
+      uuid: 'some-person-uuid-1',
+      age: 24,
+      gender: 'F',
+    },
+  };
+
+  const patient2 = {
+    uuid: 'patient-uuid-2',
+    person: {
+      uuid: 'some-person-uuid-2',
+      age: 0.5,
+      gender: 'M',
+    },
+  };
+
+  const currentWardLocation: Location = {
+    uuid: 'some-location-uuid',
+  };
+
+  const inpatientAdmissionsAtCurrentLocation: InpatientAdmission[] = [
+    {
+      patient: patient1,
+      visit: undefined,
+      encounterAssigningToCurrentInpatientLocation: undefined,
+      firstAdmissionOrTransferEncounter: undefined,
+      currentInpatientRequest: undefined,
+      currentInpatientLocation: currentWardLocation,
+    },
+    {
+      patient: patient2,
+      visit: undefined,
+      encounterAssigningToCurrentInpatientLocation: undefined,
+      firstAdmissionOrTransferEncounter: undefined,
+      currentInpatientRequest: undefined,
+      currentInpatientLocation: currentWardLocation,
+    },
+  ];
+
+  const admissionLocationFetchResponse: AdmissionLocationFetchResponse = {
+    bedLayouts: [
+      {
+        rowNumber: 1,
+        columnNumber: 1,
+        bedNumber: '1',
+        bedId: 1,
+        bedUuid: 'some-bed-uuid',
+        status: 'OCCUPIED',
+        bedType: undefined,
+        location: '',
+        patients: [patient1],
+        bedTagMaps: [],
+      },
+    ],
+    occupiedBeds: 1,
+    totalBeds: 1,
+    ward: currentWardLocation,
+  };
+
+  describe('createAndGetWardPatientGrouping', () => {
```

Comment:
This function was a bit funky and lacking in testing... I tried to start adding a few as I tried to understand what is does, but we should consider adding more.